src/transformers/generation_utils.py

```diff
@@ -1500,22 +1500,11 @@ def generate(
                     f"num_return_sequences has to be 1, but is {num_return_sequences} when doing contrastive search."
                 )
 
-            # 10. prepare logits warper: get the TopKLogitsWarper for contrastive_search
-            logits_warper = self._get_logits_warper(
-                top_k=top_k,
-                top_p=top_p,
-                typical_p=typical_p,
-                temperature=temperature,
-                num_beams=num_beams,
-                renormalize_logits=renormalize_logits,
-            )
```

Comment:
This shouldn't have been removed :)

Reply:
During the test, I found that adding the logits_warper (the TopKLogitsWarper is used) could influence the generations of contrastive search. Because the TopKLogitsWarper filters the logits of other tokens (not Top-k tokens) and calculates the `softmax`, the `model confidence` is different from the case that the TopKLogitsWarper is not used. 

So, in this case, I think the contrastive search should disable the logits_warper by default. 

Is there any solution that the TopKLogitsWarper warper is not activated for contrastive search?

Reply:
Interesting. In that case, I think we can do without the logits warper for now :)

Reply:
Okay!